front/src/applications/stdcm/components/StdcmResults/SimulationReportSheet.tsx

```diff
@@ -25,8 +24,12 @@ const getStopType = (step: StdcmPathStep, t: TFunction) => {
   return capitalizeFirstLetter(t(`stdcm:trainPath.stopType.${step.stopType}`));
 };
 
-const getArrivalTime = (step: StdcmPathStep, t: TFunction) => {
-  if (!step.isVia) {
+export const getArrivalTimes = (
+  step: StdcmPathStep,
+  { isFirstStep, isLastStep }: { isFirstStep?: boolean; isLastStep?: boolean },
+  t: TFunction
+) => {
+  if ((isFirstStep || isLastStep) && !step.isVia) {
```

Comment:
Checking that the step is not a via is enough (by definition, it means that it is either the origin or the destination). You can remove the args `isFirstStep` and `isLastStep`
```suggestion
  if (!step.isVia) {
```

Reply:
the naming of the variables made the intention unclea, change the naming i don't know if it's better now ?